semgrep/semgrep/git.py

```diff
@@ -0,0 +1,332 @@
+import subprocess
```

Comment:
Mostly copies from https://github.com/returntocorp/semgrep-action/blob/5cb31fb22ea0fffbae63b5ad0986d42caa2b562c/src/semgrep_agent/targets.py with necessary changes to work in semgrep-cli